test/jdk/javax/sound/sampled/Clip/AutoCloseTimeCheck.java

```diff
@@ -0,0 +1,128 @@
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.util.concurrent.TimeUnit;
+
+import javax.sound.SoundClip;
+import javax.sound.sampled.AudioFileFormat.Type;
+import javax.sound.sampled.AudioFormat;
+import javax.sound.sampled.AudioInputStream;
+import javax.sound.sampled.AudioSystem;
+
+import static javax.sound.sampled.AudioFormat.Encoding.PCM_SIGNED;
+import static javax.sound.sampled.AudioSystem.NOT_SPECIFIED;
+
+/**
+ * @test
+ * @bug 8202264
+ */
+public final class AutoCloseTimeCheck {
+
+    public static void main(final String[] args) throws Exception {
+        // Prepare the audio file
+        File file = new File("audio.wav");
+        try {
+            AudioFormat format =
+                    new AudioFormat(PCM_SIGNED, 44100, 8, 1, 1, 44100, false);
+            AudioSystem.write(getStream(format), Type.WAVE, file);
+        } catch (final Exception ignored) {
+            return; // the test is not applicable
+        }
+        try {
+            testSmallDelay(file);
+            testBigDelay(file);
+        } finally {
+            Files.delete(file.toPath());
+        }
+    }
+
+    /**
+     * Checks that after a big period of non-activity the clip will be closed
+     * and the "Direct Clip" thread will stop.
+     */
+    private static void testBigDelay(final File file) throws Exception {
+        SoundClip clip = SoundClip.createSoundClip(file);
+        clip.loop();
+        clip.stop();
+        sleep(20000); // 20 sec for slow systems
+        if (count() != 0) {
+            throw new RuntimeException("Thread was found");
+        }
+    }
+
+    /**
+     * Checks that after small period of non-activity the clip will not be
+     * closed and the "Direct Clip" thread will alive.
+     */
+    private static void testSmallDelay(final File file) throws IOException {
+        SoundClip clip = SoundClip.createSoundClip(file);
+        long threadID = 0;
+        // Will run the test no more than 15 seconds
+        long endtime = System.nanoTime() + TimeUnit.SECONDS.toNanos(15);
+        while (endtime - System.nanoTime() > 0) {
+            clip.loop();
+            sleep(500);
+
+            long data = count();
+            if (data != threadID) {
+                System.out.println("Playing on new thread: " + data + " at "
+                                           + new java.util.Date());
+                if (threadID == 0) {
+                    threadID = data;
+                } else {
+                    throw new RuntimeException("Thread was changed");
+                }
+            }
+
+            clip.stop();
+            sleep(500);
+        }
+    }
+
+    private static void sleep(int millis) {
+        try {
+            Thread.sleep(millis);
+        } catch (InterruptedException ignored) {
+        }
+    }
+
+    private static long count() {
+        for (final Thread t : Thread.getAllStackTraces().keySet()) {
```

Comment:
Why are stack traces are needed?

```java
Thread[]  threads = new Thread[Thread.activeCount()];
Thread.enumerate(threads);
```
would give the same estimate and use less resources.

[`Thread.enumerate`](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/lang/Thread.html#enumerate(java.lang.Thread%5B%5D)) fills the provided array with currently active threads, and [`Thread.activeCount`](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/lang/Thread.html#activeCount()) provides an estimate on the number of live threads.